tests/GltfTests.cpp

```diff
@@ -0,0 +1,150 @@
+            throw std::runtime_error("test model is empty");
+        }
+
+        // --- Begin checks ---
+        CesiumGltf::MeshPrimitive prim = gltf.model->meshes[0].primitives[0];
+        CesiumGltf::Model model = *gltf.model;
+
+        namespace gltfUtil = cesium::omniverse::GltfUtil;
+
+        CHECK(gltfUtil::hasNormals(model, prim, true) == expectedResults["hasNormals"].as<bool>());
+        CHECK(gltfUtil::hasTexcoords(model, prim, 0) == expectedResults["hasTexcoords"].as<bool>());
+        CHECK(gltfUtil::hasImageryTexcoords(model, prim, 0) == expectedResults["hasImageryTexcoords"].as<bool>());
+        CHECK(gltfUtil::hasVertexColors(model, prim, 0) == expectedResults["hasVertexColors"].as<bool>());
+        CHECK(gltfUtil::hasMaterial(prim) == expectedResults["hasMaterial"].as<bool>());
+        CHECK(gltfUtil::getDoubleSided(model, prim) == expectedResults["doubleSided"].as<bool>());
+
+        // material tests
+        if (gltfUtil::hasMaterial(prim)) {
+            CesiumGltf::Material mat = gltf.model->materials[0];
+            CHECK(gltfUtil::getAlphaMode(mat) == expectedResults["alphaMode"].as<int>());
+            CHECK(gltfUtil::getAlphaCutoff(mat) == expectedResults["alphaCutoff"].as<float>());
+            CHECK(gltfUtil::getBaseAlpha(mat) == expectedResults["baseAlpha"].as<float>());
+            CHECK(gltfUtil::getMetallicFactor(mat) == expectedResults["metallicFactor"].as<float>());
+            CHECK(gltfUtil::getRoughnessFactor(mat) == expectedResults["roughnessFactor"].as<float>());
+            CHECK(gltfUtil::getBaseColorTextureWrapS(model, mat) == expectedResults["baseColorTextureWrapS"].as<int>());
+            CHECK(gltfUtil::getBaseColorTextureWrapT(model, mat) == expectedResults["baseColorTextureWrapT"].as<int>());
+
+            CHECK(gltfUtil::getBaseColorFactor(mat) == expectedResults["baseColorFactor"].as<std::vector<float>>());
+            CHECK(gltfUtil::getEmissiveFactor(mat) == expectedResults["emissiveFactor"].as<std::vector<float>>());
+        }
+
+        // Accessor smoke tests
+        cesium::omniverse::PositionsAccessor positions;
+        cesium::omniverse::IndicesAccessor indices;
+        CHECK_NOTHROW(positions = gltfUtil::getPositions(model, prim));
+        CHECK_NOTHROW(indices = gltfUtil::getIndices(model, prim, positions));
+        if (gltfUtil::hasNormals(model, prim, true)) {
+            CHECK_NOTHROW(gltfUtil::getNormals(model, prim, positions, indices, false));
+        }
+        if (gltfUtil::hasVertexColors(model, prim, 0)) {
+            CHECK_NOTHROW(gltfUtil::getVertexColors(model, prim, 0));
+        }
+        if (gltfUtil::hasTexcoords(model, prim, 0)) {
+            CHECK_NOTHROW(gltfUtil::getTexcoords(model, prim, 0));
+        }
+        if (gltfUtil::hasImageryTexcoords(model, prim, 0)) {
+            CHECK_NOTHROW(gltfUtil::getImageryTexcoords(model, prim, 0));
+        }
+        CHECK_NOTHROW(gltfUtil::getExtent(model, prim));
```

Comment:
These functions aren't expected to throw. A better way to check the validity of an accessor would be to check if the size() is greater than 0, and for the extent to check that it's not `nullopt`.